File: src/prod/src/ServiceModel/management/FaultAnalysisService/Chaos/ChaosStatus.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

namespace Management
{
    namespace FaultAnalysisService
    {
        using namespace Common;

        namespace ChaosStatus
        {
            void WriteToTextWriter(Common::TextWriter & w, Enum const & val)
            {
                switch (val)
                {
                case Running:
                    w << L"Running";
                    return;
                case Stopped:
                    w << L"Stopped";
                    return;
                default:
                    w << wformatString("Unknown ChaosStatus value {0}", static_cast<int>(val));
                }
            }

            Common::ErrorCode FromPublicApi(FABRIC_CHAOS_STATUS const & publicStatus, __out Enum & status)
            {
                switch (publicStatus)
                {
                case FABRIC_CHAOS_STATUS_RUNNING:
                    status = Running;
                    break;
                case FABRIC_CHAOS_STATUS_STOPPED:
                    status = Stopped;
                    break;
                default:
                    return ErrorCode(ErrorCodeValue::InvalidArgument, wformatString("{0}", FASResource::GetResources().ChaosStatusUnknown));
                }

                return ErrorCode(ErrorCodeValue::Success);
            }

            FABRIC_CHAOS_STATUS ToPublicApi(Enum const & status)
            {
                switch (status)
                {
                case Running:
                    return FABRIC_CHAOS_STATUS_RUNNING;
                case Stopped:
                    return FABRIC_CHAOS_STATUS_STOPPED;
                default:
                    return FABRIC_CHAOS_STATUS_INVALID;
                }
            }

            wstring ChaosStatus::ToString(ChaosStatus::Enum const & val)
            {
                switch (val)
                {
                    case(ChaosStatus::Running):
                        return L"Running";
                    case(ChaosStatus::Stopped) :
                        return L"Stopped";
                    default:
                        return L"Invalid";
                }
            }
        }
    }
}

File: src/prod/src/managed/Api/src/System/Fabric/Chaos/DataStructures/ChaosStatus.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric.Chaos.DataStructures
{
    using System.Fabric.Interop;

    /// <summary>
    /// <para>Represents the task that Chaos is presently executing.</para>
    /// </summary>
    public enum ChaosStatus
    {
        /// <summary>
        /// <para>Indicates that the current task is invalid.</para>
        /// </summary>
        None = NativeTypes.FABRIC_CHAOS_STATUS.FABRIC_CHAOS_STATUS_INVALID,

        /// <summary>
        /// <para>Indicates that Chaos is executing actions as part of fault inducing.</para>
        /// </summary>
        Running = NativeTypes.FABRIC_CHAOS_STATUS.FABRIC_CHAOS_STATUS_RUNNING,

        /// <summary>
        /// <para>Indicates that Chaos has finished inducing faults and now it is validating if the Cluster is stable and healthy.</para>
        /// </summary>
        Stopped = NativeTypes.FABRIC_CHAOS_STATUS.FABRIC_CHAOS_STATUS_STOPPED
    }
}